examples/webserver/threads/src/main/java/io/helidon/examples/webserver/threads/ThreadService.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.examples.webserver.threads;
+
+import java.lang.System.Logger.Level;
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
+
+import io.helidon.common.configurable.ThreadPoolSupplier;
+import io.helidon.config.Config;
+import io.helidon.http.Status;
+import io.helidon.webclient.api.ClientResponseTyped;
+import io.helidon.webclient.api.WebClient;
+import io.helidon.webserver.http.HttpRules;
+import io.helidon.webserver.http.HttpService;
+import io.helidon.webserver.http.ServerRequest;
+import io.helidon.webserver.http.ServerResponse;
+
+class ThreadService implements HttpService {
+
+    private static final System.Logger LOGGER = System.getLogger(ThreadService.class.getName());
+    private static final Random rand = new Random(System.currentTimeMillis());
+
+    // ThreadPool of platform threads.
+    private static ExecutorService platformExecutorService;
+    // Executor of virtual threads.
+    private static ExecutorService virtualExecutorService;
+
+    /**
+     * The config value for the key {@code greeting}.
+     */
+
+    ThreadService() {
+        this(Config.global().get("app"));
+    }
+
+    ThreadService(Config appConfig) {
+        /*
+         * We create two executor services. One is a thread pool of platform threads.
+         * The second is a virtual thread executor service.
+         * See `application.yaml` for configuration of each of these.
+         */
+        ThreadPoolSupplier platformThreadPoolSupplier = ThreadPoolSupplier.builder()
+                .config(appConfig.get("application-platform-executor"))
+                .build();
+        platformExecutorService = platformThreadPoolSupplier.get();
+
+        ThreadPoolSupplier virtualThreadPoolSupplier = ThreadPoolSupplier.builder()
+                .config(appConfig.get("application-virtual-executor"))
+                .build();
+        virtualExecutorService = virtualThreadPoolSupplier.get();
```

Comment:
`virtualThreadPoolSupplier` sounds really bad

Reply:
Renamed to `virtualThreadSupplier`